src/entity.zig

```diff
@@ -226,33 +228,31 @@ pub const ClientEntityManager = struct {
 		}
 	}
 
-	pub fn serverUpdate(time: i16, data: []const u8) void {
+	pub fn serverUpdate(time: i16, reader: *BinaryReader) !void {
 		mutex.lock();
 		defer mutex.unlock();
+		if(reader.remaining.len%(4 + 24 + 12 + 24) != 0) return error.corrupted;
```

Comment:
I do not think we still need this check, it is also covered by the `try`s below.

Reply:
Ok. Mind that this changes semantics of this function. If message is corrupted it can have partial effect on entities, since only last segment will be discarded.

Reply:
If a message is corrupted, the player gets disconnected anyways.

Reply:
The main reason why this can happen, is if the player is joining with an old version of the game and there were some changes to the protocol in the meantime.